DistributedSystem/Data-Consistency-Model.md

```diff
@@ -0,0 +1,33 @@
+# Data Consistency Model In Distributed System
+
+
+Một trong những thành phần chính tạo ra một hệ thống thông tin, đó là các đối tượng dữ liệu. Các đối tượng dữ liệu trên hệ thống thông tin có thể tồn tại dưới những dạng khác nhau, như các cơ sở dữ liệu, các file hoặc thậm chí là các đối tượng nằm trong bộ nhớ RAM của một process. Tuy nhiên, cách thức sử dụng và quản lý các đối tượng dữ liệu trong một hệ thống phân tán có sự khác biệt với một hệ thống đơn tiến trình truyền thống. Khác với các hệ thống đơn tiến trình truyền thống thường chỉ lưu trữ và quản lý các đối tượng dữ liệu dưới dạng một thực thể duy nhất, trong một hệ thống phân tán, một đối tượng dữ liệu có thể tồn tại dưới dạng nhiều thực thể giống nhau - được gọi là các bản sao (replica) của đối tượng.
+
+Việc tạo ra nhiều bản sao cho một đối tượng dữ liệu có rất nhiều ưu điểm, ví dụ như cho phép tăng hiệu năng tổng thể của hệ thống, do việc trên hệ thống việc có nhiều bản sao cho phép cung cấp nhiều tài nguyên hơn. Đồng thời, việc có nhiều bản sao của một đối tượng dữ liệu trên hệ thống cho phép hệ thống có khả năng chống lỗi tốt hơn, do khi sự cố xảy ra trên một thành phần nào đó, thì các bản sao khác của các đối tượng dữ liệu vẫn tồn tại ở trên các thành phần khác trong hệ thống, do đó hệ thống vẫn có khả năng phục hồi lại dữ liệu cho người dùng.
+
+Chúng ta có thể thấy cơ chế sao lưu (data replication) cung cấp rất nhiều lợi ích cho hệ thống phân tán. Tuy nhiên,đi cùng với các lợi ích mang lại, việc một đối tượng dữ liệu có nhiều bản sao trong hệ thống phân tán cũng đặt ra cho chúng ta một số vấn đề cần giải quyết. Đó là trong hệ thống phân tán, khi nhiều bản sao đồng thời phục vụ cho nhiều thực thể (các tiến trình và cácclient) khác nhau và bị các thực thể này thay đổi nội dung, làm sao để chúng ta xác định được tại một thời điểm xác định, đối tượng dữ liệu cũng như các bản sao của nó trên hệ thống sẽ có giá trị là gì ?
+
+Ví dụ, trong hệ thống phân tán có thể xảy ra sự kiện cùng 1 lúc nhiều client yêu cầu thay đổi nội dung của một replica:
```

Comment:
Anh ơi, em nghĩ là mình nên trình bày cái định lý CAP sau khi đã giới thiệu về 2 mô hình ACID và BASE anh ạ. Vì trong này ông phát biểu định lý này cũng nói tới CAP sau 2 mô hình kia ạ.

http://pld.cs.luc.edu/courses/353/spr11/notes/brewer_keynote.pdf

Với cả trong giải thích của wikipedia, thì trong CAP, cái consistency nó là một tính chất cái cụ thể chứ không mang tính tổng quá nữa ạ:

Consistency: Every read receives the most recent write or an error